robot-name/robot_name_test.rb

```diff
@@ -29,6 +29,7 @@ def test_reset_name
     robot.reset
     name2 = robot.name
     assert name != name2
+    assert_equal robot.name = name2
```

Comment:
Did you try running this assertion? The api of `assert_equal` takes two arguments, plus an optional argument for the failure message.

Since the code does `name2 = robot.name` on line 30, it doesn't really make sense to assign the same value back with `robot.name = name2`.